### PR TITLE
Implemented loading of country unit colours

### PR DESCRIPTION
**Pull request by BetterBite**
Country unit colours added to Country struct and loaded in from the country_colours.txt file

### CODE REVIEW

File: src/openvic-simulation/country/Country.cpp [hunk 1]
```diff
@@ -24,17 +24,33 @@ CountryParty::CountryParty(
...
+	secondary_unit_colour { new_secondary_unit_colour },
+	tertiary_unit_colour { new_tertiary_unit_colour } 
+	{}
 
 bool CountryManager::add_country(
 	std::string_view identifier, colour_t colour, GraphicalCultureType const* graphical_culture,
 	IdentifierRegistry<CountryParty>&& parties, Country::unit_names_map_t&& unit_names, bool dynamic_tag,
-	Country::government_colour_map_t&& alternative_colours
+	Country::government_colour_map_t&& alternative_colours, 
+	colour_t primary_unit_colour, colour_t secondary__unit_colour, colour_t tertiary__unit_colour
```
Comment (BrickPi) [inline src/openvic-simulation/country/Country.cpp:53]: What's the rationale behind the double underscores?

File: src/openvic-simulation/dataloader/Dataloader.cpp [hunk 2]
```diff
@@ -808,6 +808,7 @@ bool Dataloader::load_defines(GameManager& game_manager) {
 	static constexpr std::string_view buildings_file = "common/buildings.txt";
 	static constexpr std::string_view bookmark_file = "common/bookmarks.txt";
 	static constexpr std::string_view countries_file = "common/countries.txt";
+	static constexpr std::string_view country_colours_file = "common/country_colors.txt"; // TODO - Load this
```
Comment (BrickPi) [inline src/openvic-simulation/dataloader/Dataloader.cpp:811]: Todo done :)

**Review by BrickPi — COMMENTED** [review]
LGTM, one comment and a touch up. Should squash down to a single commit when ready to merge.

File: src/openvic-simulation/country/Country.cpp (hunk 1 above)
Comment (BetterBite) [reply src/openvic-simulation/country/Country.cpp:53]: That was an error when i was creating the variables, woops

Comment (BetterBite) [reply src/openvic-simulation/country/Country.cpp]: fixed it

File: src/openvic-simulation/dataloader/Dataloader.cpp (hunk 2 above)
Comment (BetterBite) [reply src/openvic-simulation/dataloader/Dataloader.cpp]: removed

**Comment by BrickPi** [comment]
Looks good, now just squash it and then we can merge it.

**Comment by BetterBite** [comment]
Squashed

File: src/openvic-simulation/country/Country.hpp [hunk 3]
```diff
@@ -63,11 +63,16 @@ namespace OpenVic {
 		const unit_names_map_t PROPERTY(unit_names);
 		const bool PROPERTY_CUSTOM_PREFIX(dynamic_tag, is);
 		const government_colour_map_t PROPERTY(alternative_colours);
+		colour_t PROPERTY(primary_unit_colour);
+		colour_t PROPERTY(secondary_unit_colour);
+		colour_t PROPERTY(tertiary_unit_colour);
```
Comment (Spartan322) [inline src/openvic-simulation/country/Country.hpp:68]: Shouldn't these be const?

**Comment by Spartan322** [comment]
This is not a proper rebase.

File: src/openvic-simulation/country/Country.hpp (hunk 3 above)
Comment (BetterBite) [reply src/openvic-simulation/country/Country.hpp:68]: Cannot be const as it is added outside the constructor

**Comment by BetterBite** [comment]
> This is not a proper rebase.

This is my first time actually doing any sort of rebase so I apologise if i whiffed something

**Comment by BetterBite** [comment]
fixed into a proper rebase (I hope)

**Review by BrickPi — APPROVED** [review]
LGTM, first commit 💪

**Pull request merged**